test/test.ts

```diff
@@ -432,6 +436,245 @@ describe("window.codePush", function() {
         });
     });
 
+    describe("#localPackage.installOnNextResume.norevert", function() {
+
+        after(() => {
+            cleanupScenario();
+        });
+
+        before(() => {
+            return setupScenario(ScenarioInstallOnNextResume);
+        });
+
+        var getMockResponse = (): su.CheckForUpdateResponseMock => {
+            var updateReponse = createMockResponse();
+            updateReponse.downloadURL = serverUrl + "/download";
+            return updateReponse;
+        };
+
+        it("should handle install (on resume)", function(done) {
+
+            mockResponse = { updateInfo: getMockResponse() };
+
+            setupUpdateProject(UpdateDeviceReady, "Update 1")
+                .then<string>(() => { return projectManager.createUpdateArchive(updatesDirectory, targetPlatform); })
+                .then<void>((updatePath: string) => {
+                    var deferred = Q.defer<void>();
+                    mockUpdatePackagePath = updatePath;
+                    testMessageCallback = verifyMessages([su.TestMessage.UPDATE_INSTALLED], deferred);
+                    console.log("Running project...");
+                    projectManager.runPlatform(testRunDirectory, targetPlatform, true, targetEmulator);
+                    return deferred.promise;
+                })
+                .then<void>(() => {
+                    /* "resume" the application - run it again */
+                    var deferred = Q.defer<void>();
+                    testMessageCallback = verifyMessages([su.TestMessage.DEVICE_READY_AFTER_UPDATE], deferred);
+                    console.log("Running project...");
+                    var emulatorManager = targetPlatform.getOptionalEmulatorManager();
+                    if (emulatorManager) {
+                        emulatorManager.endRunningApplication(TestNamespace)
+                            .then(() => emulatorManager.launchInstalledApplication(TestNamespace));
```

Comment:
Since you're not returning a promise you need to terminate a promise chain with done().

From https://github.com/kriskowal/q/wiki/API-Reference#promisedoneonfulfilled-onrejected-onprogress:

"The Golden Rule of  done  vs.  then  usage is: either  return  your promise to someone else, or if the chain ends with you, call  done  to terminate it. Terminating with  catch  is not sufficient because the catch handler may itself throw an error."